addons/payment/tests/test_multicompany_flows.py

```diff
@@ -35,7 +35,11 @@ def setUpClass(cls):
             'group_ids': [Command.link(cls.group_user.id)],
         })
 
-        cls.provider_company_b = cls._prepare_provider(company=cls.company_b)
+        cls.provider_company_b = cls.env['payment.provider'].search(
```

Comment:
Shouldn't we overwrite the base `cls.provider` and update the multi-company tests accordingly, since we'll never use the provider from company A in this context?
```suggestion
        cls.provider = cls.env['payment.provider'].search(
```